contracts/integrations/contracts/dydx/ZeroExV3MultiOrderExchangeWrapper.sol

```diff
@@ -65,22 +68,35 @@ contract ZeroExV2MultiOrderExchangeWrapper is
 
     // ============ State Variables ============
 
-    // address of the ZeroEx V2 Exchange
+    // msg.senders that will use this contract's WETH balance to pay protocol fees
+    mapping (address => bool) public TRUSTED_MSG_SENDER;
+
+    // address of the ZeroEx V3 Exchange
     address public ZERO_EX_EXCHANGE;
 
-    // address of the ZeroEx V2 ERC20Proxy
+    // address of the ZeroEx ERC20Proxy
     address public ZERO_EX_TOKEN_PROXY;
 
+    // address of the WETH token contract
+    address public WETH_TOKEN;
+
     // ============ Constructor ============
 
     constructor(
         address zeroExExchange,
-        address zeroExProxy
+        address zeroExProxy,
+        address weth,
+        address[] memory trustedMsgSenders
     )
         public
     {
         ZERO_EX_EXCHANGE = zeroExExchange;
         ZERO_EX_TOKEN_PROXY = zeroExProxy;
+        WETH_TOKEN = weth;
+
+        for (uint256 i = 0; i < trustedMsgSenders.length; i++) {
```

Comment:
It would probably be good to add a `addTrustedAddress` function so that new addresses can be whitelisted without redeploying this contract.